static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+  const e_y = goodi.map((ind) => kwa.e_y[ind]);
+
+  const tmin = Math.min.apply(null, t_data);
+  const t = add(t_data, -tmin);
+  const tbase = Math.max.apply(null, t);
+
+  i = t.length;
+  const nt = t.length;
+  // eslint-disable-next-line no-plusplus
+  while (i--) {
+    w[i] = e_y ? 1 / e_y[i] / e_y[i] : 1;
+    wsum += w[i];
+  }
+
+  // normalize weights, now "wsum=1"
+  i = nt;
+  // eslint-disable-next-line no-plusplus
+  while (i--) w[i] /= wsum;
+
+  const ymean = dot(w, y_data);
+  const y = add(y_data, -ymean);
+  const wy = dotMultiply(w, y);
+  const YY = dot(wy, y); // Eq.(10), variance for the zero mean data
+
+  const df = 1 / tbase / ofac; // frequency sampling depends on the time span, default for start frequency
+  const delta_t = tbase / (nt - 1); // mean time sampling
+  const fbeg = parseFloat(kwargs.fbeg || df);
+  const fend = parseFloat(kwargs.fend || (0.5 * ofac) / delta_t);
+  const nf = Math.floor((fend - fbeg) / df) + 1; // size of frequency grid
+
+  const f = new Array(nf);
+  const p = new Array(nf);
+
+  // eslint-disable-next-line no-plusplus
+  for (k = 0; k < nf; k++) {
+    f[k] = fbeg + k * df;
+    omega = twopi * f[k];
+    C = 0;
+    S = 0;
+    YC = 0;
+    YS = 0;
+    CC = 0;
+    SS = 0;
+    CS = 0;
+    // eslint-disable-next-line no-plusplus
+    for (i = 0; i < nt; i++) {
+      wi = w[i];
+      cosx = Math.cos(omega * t[i]);
+      sinx = Math.sin(omega * t[i]);
+      if (!kwargs.ls) {
+        C += wi * cosx; // Eq.(8)
+        S += wi * sinx; // Eq.(9)
+      }
+      YC += wy[i] * cosx; // Eq.(11) simplifies, since Y should be 0 (mean was subtracted)
+      YS += wy[i] * sinx; // Eq.(12)   -"-
+      CC += wi * cosx * cosx; // Eq.(13)
+      CS += wi * cosx * sinx; // Eq.(15)
+    }
+    SS = 1 - CC; // Eq.(14)
+    SS -= S * S; // Eq.(14)
+    CC -= C * C; // Eq.(13)
+    CS -= C * S; // Eq.(15)
+    D = CC * SS - CS * CS; // Eq.(6)
+
+    // A[k] = (YC*SS-YS*CS) / D
+    // B[k] = (YS*CC-YC*CS) / D
+    // off[k] = -A[k]*C-B[k]*S
+
+    p[k] =
+      ((SS * YC * YC) / D + (CC * YS * YS) / D - (2 * CS * YC * YS) / D) / YY; // Eq.(5)
+    if (p[k] > p[kbest]) {
+      kbest = k;
+    }
+  }
+  return { p, f, k: kbest, fbest: f[kbest], tbase };
+}
+
+const Periodogram = () => {
+  const classes = useStyles();
+  const { handleSubmit, getValues, control } = useForm();
+  const { id } = useParams();
+  const dispatch = useDispatch();
+  const photometry = useSelector((state) => state.photometry[id]);
+  const [bestp, setBestp] = useState(null);
+  const [run, setRun] = useState(false);
+  const [plotted, setPlotted] = useState(false);
+
+  const [instruments, setInstruments] = useState([]);
+  const [filters, setFilters] = useState([]);
+  const [mjdmin, setMjdmin] = useState(0);
+  const [mjdmax, setMjdmax] = useState(70000);
+
+  const [params, setParams] = useState({
+    instrument: "P60 Camera",
+    filter: "ztfr",
+    ofac: "20",
+  });
+
+  const placeholder = (
```

Comment:
Prefer moving JSX literal to render in-line in main return statement below, doing away with this variable

Reply:
Done.